# Predicting Usage of a Video Game Research Server

### Data Description<br>
players:It has 7 variables.<br>
      experience:the player's game experience<br>
      subscribe:whether the player subscribe the game<br>
      hashedEmail:hashed email<br>
      played_hours:player's gaming time<br>
      name：player's name<br>
      gender:player's gender<br>
      Age:player's age</p>
sessions: It has 5 variables<br>
      hashedEmail:hashed email<br>
      start_time:start time of the session<br>
      end_time:end time of the session<br>
      original_start_time:original start time of the session<br>
      original_end_time:original end time of the session<br>

### Questions


Question 1: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?</p>
Can experience,played_hours,gender and age predict subscribe in players.csv?

### Exploratory Data Analysis and Visualization

In [1]:
library(tidyverse)
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimite

In [10]:
players_mean <- summarize(players,played_hours=mean(played_hours,na.rm = TRUE),
          Age=mean(Age,na.rm = TRUE))
players_mean

played_hours Age     
1 5.845918     21.13918

| quantitative variable | mean value |
| -------- | ------- |
| played_hours	 | 5.845918 |
| Age | 21.13918 |


### Methods and Plan

I use "players.csv".Because the question focus on player characteristics.The "players.csv" dataset can provide related information.</p>
Then I can use Classification to build different models and compare the accuracy.